src/server/pfs/server/env.go

```diff
@@ -13,11 +13,16 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/internal/serviceenv"
 	"github.com/pachyderm/pachyderm/v2/src/internal/task"
 	txnenv "github.com/pachyderm/pachyderm/v2/src/internal/transactionenv"
+	"github.com/pachyderm/pachyderm/v2/src/internal/transactionenv/txncontext"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
 	authserver "github.com/pachyderm/pachyderm/v2/src/server/auth"
-	ppsserver "github.com/pachyderm/pachyderm/v2/src/server/pps"
 	etcd "go.etcd.io/etcd/client/v3"
 )
 
+type PPS interface {
```

Comment:
This is kind of a time-saving hack that I don't really support.  There should be an `type PipelineInspector interface { InspectPipelineInTransaction(...) ... }` that the PFS server should accept, and it just so happens that the PPS server implements that interface.  As soon as you make it a catch-all, it becomes very unwieldy.

Reply:
done.